weldx/core.py

```diff
@@ -385,6 +387,68 @@ def __repr__(self):
             )
         return representation + f"Units:\n\t{self.units}\n"
 
+    def _interp_time_discrete(
+        self, time: Union[pd.TimedeltaIndex, pint.Quantity]
+    ) -> xr.DataArray:
+        """Interpolate the time series if its data is composed of discrete values.
+
+        See `interp_time` for interface description.
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            if isinstance(time, pint.Quantity):
+                time = ut.to_pandas_time_index(time)
+            if not isinstance(time, pd.TimedeltaIndex):
+                raise ValueError(
+                    '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
+                )
+            # constant values are also treated by this branch
+            if self._data.attrs["interpolation"] == "linear" or self.shape[0] == 1:
+                return ut.xr_interp_like(
+                    self._data,
+                    {"time": time},
+                    assume_sorted=False,
+                    broadcast_missing=False,
+                )
+
+            dax = self._data.reindex({"time": time}, method="ffill")
+            return dax.fillna(self._data[0])
```

Comment:
since the outside branch was the "step" interpolation anyways (I think), can we simplify this to just use `util.xr_interp_like` now for all interpolation methods?

and if we do that it should also allow us to enable all xarray supported interpolation methods as well?

Reply:
I added the other interpolation methods manually but I think it would be nicer to extract them somehow from xarray. A quick search didn't bring up a list that I could use to update ours. Didn't search very hard though. If you know where to find one, let me know and I will add it.

Reply:
I couldn't find a list from xarray (also didn't look too hard) but I think it's fine now this way :)